src/hotspot/share/runtime/vmThread.cpp

```diff
@@ -467,20 +468,20 @@ void VMThread::loop() {
 
   SafepointSynchronize::init(_vm_thread);
 
+  assert(Thread::current()->is_VM_thread(), "Must be the VM thread");
```

Comment:
This is unnecessary, we don't need to guard against accidental calls to VMThread::loop by other threads!

Reply:
Removed